test/e2e/autoscaling_utils.go

```diff
@@ -114,14 +116,37 @@ func (rc *ConsumingRC) sendOneConsumeCPUrequest(milicores int, durationSec int)
 	expectNoError(err)
 }
 
-func (rc *ConsumingRC) CleanUp() {
+func (rc *ResourceConsumer) GetReplicas() int {
+	replicationController, err := rc.framework.Client.ReplicationControllers(rc.framework.Namespace.Name).Get(rc.name)
+	expectNoError(err)
+	if replicationController == nil {
+		Failf(rcIsNil)
+	}
+	return replicationController.Status.Replicas
+}
+
+func (rc *ResourceConsumer) WaitForReplicas(desiredReplicas int) {
+	timeout := 10 * time.Minute
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(20 * time.Second) {
+		if desiredReplicas == rc.GetReplicas() {
+			Logf("Replication Controller current replicas number is equal to desired replicas number: %d", desiredReplicas)
+			return
+		} else {
+			Logf("Replication Controller current replicas number %d waiting to be %d", rc.GetReplicas(), desiredReplicas)
+		}
+	}
+	Failf("timeout waiting %v for pods size to be %d", timeout, desiredReplicas)
+}
+
+func (rc *ResourceConsumer) CleanUp() {
 	rc.stop <- 0
 	expectNoError(DeleteRC(rc.framework.Client, rc.framework.Namespace.Name, rc.name))
 	expectNoError(rc.framework.Client.Services(rc.framework.Namespace.Name).Delete(rc.name))
+	expectNoError(rc.framework.Client.Experimental().HorizontalPodAutoscalers(rc.framework.Namespace.Name).Delete(rc.name, api.NewDeleteOptions(0)))
```

Comment:
@socaa This is a very bad pattern when you clean up here resources created by someone else. Please move this call to test/e2e/horizontal_pod_autoscaling.go